client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you.tsx

```diff
@@ -54,14 +58,29 @@ const MarketplaceThankYou = ( {
 		thankYouHeaderAction,
 	] = usePluginsThankYouData( pluginSlugs );
 	const [
+		themesList,
 		themesSection,
 		allThemesFetched,
 		themesGoBackSection,
 		themeTitle,
 		themeSubtitle,
 		themesProgressbarSteps,
 		isAtomicNeededForThemes,
-	] = useThemesThankYouData( themeSlugs, isOnboardingFlow );
+	] = useThemesThankYouData( themeSlugs, isOnboardingFlow, continueWithPluginBundle );
+
+	const firstTheme = themesList[ 0 ];
+	useEffect( () => {
+		if ( firstTheme && styleVariationSlug ) {
+			const styleVariation = firstTheme.style_variations.filter(
+				( variation: { slug: string } ) => variation.slug === styleVariationSlug
+			);
+			dispatch(
+				setThemePreviewOptions( firstTheme.id, null, null, {
+					styleVariation: styleVariation[ 0 ],
+				} )
+			);
+		}
+	}, [ dispatch, firstTheme, styleVariationSlug ] );
```

Comment:
Why are we using the first theme from the list? When we dispatch styleVariation[ 0 ] might not exist. I think we could simplify this by:
```

useEffect(() => {
  if (firstTheme && styleVariationSlug) {
    const styleVariation = firstTheme.style_variations.find(variation => variation.slug === styleVariationSlug);
    if (styleVariation) {
      dispatch(setThemePreviewOptions(firstTheme.id, null, null, { styleVariation }));
    }
  }
}, [dispatch, firstTheme, styleVariationSlug]);
```